packages/cli/src/workflows/workflows.services.ee.ts

```diff
@@ -90,9 +92,51 @@ export class EEWorkflowsService extends WorkflowsService {
 			workflow.sharedWith?.push({ id, email, firstName, lastName });
 		});
 
-		// @ts-ignore
 		delete workflow.shared;
 
+		const userCredentials = await EECredentials.getAll(currentUser);
+		const credentialIdsUsedByWorkflow = new Set<number>();
+		workflow.nodes.forEach((node) => {
+			if (!node.credentials) {
+				return;
+			}
+			Object.keys(node.credentials).forEach((credentialType) => {
+				const credential = node.credentials?.[credentialType];
+				if (!credential?.id) {
+					return;
+				}
+				const credentialId = parseInt(credential.id, 10);
+				credentialIdsUsedByWorkflow.add(credentialId);
+			});
+		});
+		const workflowCredentials = await EECredentials.getMany({
+			where: {
+				id: In(Array.from(credentialIdsUsedByWorkflow)),
+			},
+		});
+
+		return EEWorkflowsService.addCredentialsToWorkflow(
+			workflow,
+			workflowCredentials,
+			userCredentials,
+		);
+	}
+
+	static addCredentialsToWorkflow(
+		workflow: WorkflowWithSharingsAndCredentials,
+		workflowCredentials: ICredentialsDb[],
+		userCredentials: ICredentialsDb[],
+	): WorkflowWithSharingsAndCredentials {
```

Comment:
`addOwnerSharingsAndCredentials` suggests that this method is doing too much. I would try to avoid major service methods calling each other, i.e. breaking off this inner call and making separate calls from the controller. This allows us to keep an overview of what is going on at the controller level, and to keep methods focused.

Suggestion, not a blocker.